.github/workflows/build-toolchain.yml

```diff
@@ -0,0 +1,35 @@
+name: Build toolchain
+
+on:
+  workflow_dispatch:
+
+jobs:
+  build-macos:
+    name: build-macos-toolchain
+    runs-on: macos-12
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      - name: Install denependecies
+        run: |
+          brew install bash coreutils autoconf automake genext2fs make wget gnu-sed
+          echo "$(brew --prefix make)/libexec/gnubin" >> $GITHUB_PATH
+          echo "$(brew --prefix gnu-sed)/libexec/gnubin" >> $GITHUB_PATH
+
+      - name: Build toolchain
+        run: |
+          mkdir ~/toolchain-macos-i386
+          git rev-parse --short HEAD > ~/toolchain-macos-i386/git-version
+          cd phoenix-rtos-build/toolchain/
+          ./build-toolchain.sh i386-pc-phoenix $HOME/toolchain-macos-i386/
+          rm -rf ~/toolchain-macos-i386/*.patch ~/toolchain-macos-i386/binutils-* ~/toolchain-macos-i386/gcc-*
+          tar -C $HOME -cvf ~/toolchain-macos-i386.tar toolchain-macos-i386
```

Comment:
I would appreciate comment why compression here is not needed / useless

Reply:
done